components/payments/internal/connectors/plugins/public/stripe/client/payments.go

```diff
@@ -0,0 +1,51 @@
+package client
+
+import (
+	"context"
+
+	"github.com/stripe/stripe-go/v79"
+)
+
+var (
+	expandSource                    = "data.source"
+	expandSourceCharge              = "data.source.charge"
+	expandSourceDispute             = "data.source.dispute"
+	expandSourcePayout              = "data.source.payout"
+	expandSourceRefund              = "data.source.refund"
+	expandSourceTransfer            = "data.source.transfer"
+	expandSourcePaymentIntent       = "data.source.payment_intent"
+	expandSourceRefundPaymentIntent = "data.source.refund.payment_intent"
+)
+
+func (c *client) GetPayments(ctx context.Context, accountID *string, lastID *string, pageSize int64) ([]*stripe.BalanceTransaction, bool, error) {
+	// TODO
+	//	f := connectors.ClientMetrics(ctx, "stripe", "list_accounts")
+	//	now := time.Now()
+	//	defer f(ctx, now)
+	filters := stripe.ListParams{
+		Limit: &pageSize,
+	}
+	if lastID == nil {
+		filters.StartingAfter = lastID
+	}
+
+	expand := []*string{
+		&expandSource,
+		&expandSourceCharge,
+		&expandSourceDispute,
+		&expandSourcePayout,
+		&expandSourceRefund,
+		&expandSourceTransfer,
+		&expandSourcePaymentIntent,
+		&expandSourceRefundPaymentIntent,
+	}
+
+	itr := c.balanceTransactionClient.List(&stripe.BalanceTransactionListParams{
```

Comment:
Actually, we're gonna have a problem, since the balance transactions are started from the most recent, you will fetch the whole history without problems, but when new payments are coming you will never fetch them since they are on the other side of the list

Reply:
Ok I see now that the original implementation was using `created[lt]` to fetch balance transactions. Will see how I can replicate that with the SDK.

Reply:
I've updated it to fetch transactions, accounts and external accounts in ASC order